### PR TITLE
feat(autoware_map_msgs): add selected map loading

### PR DESCRIPTION
**Pull request by Shin-kyoto**
## Description

Add autoware_map_msgs to integrate a new interface for selected map loading. 
See https://github.com/orgs/autowarefoundation/discussions/2812#discussioncomment-5540316 for detail.

## Related links

The next step of this PR would be to create a selected map loading functionality in map_loader. Here's how it's going to look like: https://github.com/autowarefoundation/autoware.universe/pull/3286

- https://github.com/autowarefoundation/autoware.universe/pull/3286
   - add selected map loading support
- https://github.com/autowarefoundation/autoware_launch/pull/285
   - modify config for pointcloud_map_loader

## Notes for reviewers

- Are there any better names for each messages/service?

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: autoware_map_msgs/msg/PointCloudMapMetaData.msg [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Header
+std_msgs/Header header
+
+PointCloudMapCellMetaDataWithID[] pointcloud_map_metadata
```
Comment (kminoda) [inline autoware_map_msgs/msg/PointCloudMapMetaData.msg:4]: How about making this plural, or sound like plural? Something like:
```suggestion
PointCloudMapCellMetaDataWithID[] metadata_list
```

File: autoware_map_msgs/README.md [hunk 2]
```diff
@@ -29,3 +33,7 @@ Let $X_0$ be a set of PCD map ID that the client node has, $X_1$ be a set of PCD
 ( $x \in A\backslash B \iff x \in A \wedge x \notin B$ )
 
 <img src="./media/differential_area_loading.gif" alt="drawing" width="400"/>
+
+## GetSelectedPointCloudMap.srv
+
+Given IDs query that the client node already has, the response is expected to contain the PCD maps (each of which attached with unique ID).
```
Comment (kminoda) [inline autoware_map_msgs/README.md:39]: `the client node already has` part should be clarified. Maybe you should also mention that the client is expected to receive `PointCloudMapCellMetaDataWithID.msg` metadata beforehand to use this interface.

File: autoware_map_msgs/msg/PointCloudMapMetaData.msg (hunk 1 above)
Comment (Shin-kyoto) [reply autoware_map_msgs/msg/PointCloudMapMetaData.msg]: Thanks!! 
I made this plural in https://github.com/autowarefoundation/autoware_msgs/pull/57/commits/ba347e2ee444361930691ca8a56b21fe53bf8c58

File: autoware_map_msgs/README.md (hunk 2 above)
Comment (Shin-kyoto) [reply autoware_map_msgs/README.md]: Thanks!!
I added that information in https://github.com/autowarefoundation/autoware_msgs/pull/57/commits/565d5c76bce909a0a241c00e1a10c294c715d436

**Review by kminoda — APPROVED** [review]
LGTM

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**